mesonbuild/mesonlib.py

```diff
@@ -1332,57 +1409,63 @@ def detect_subprojects(spdir_name, current_dir='', result=None):
                 result[basename] = [trial]
     return result
 
+
 def substring_is_in_list(substr: str, strlist: T.List[str]) -> bool:
     for s in strlist:
         if substr in s:
             return True
     return False
 
-class OrderedSet(collections.abc.MutableSet):
+
+class OrderedSet(T.MutableSet[_T]):
     """A set that preserves the order in which items are added, by first
     insertion.
     """
-    def __init__(self, iterable=None):
-        self.__container = collections.OrderedDict()
+    def __init__(self, iterable: T.Optional[T.Iterable[_T]] = None):
+        # typing.OrderedDict is new in 3.7.2, so we can't use that, but we can
+        # use MutableMapping, which is fine in this case.
+        self.__container = collections.OrderedDict()  # type: T.MutableMapping[_T, None]
         if iterable:
             self.update(iterable)
 
-    def __contains__(self, value):
+    def __contains__(self, value: T.Any) -> bool:
```

Comment:
```suggestion
    def __contains__(self, value: object) -> bool:
```

Reply:
Interesting, I didn't realize that `Any` and `object` were not equivalent. For anyone else curious, Any basically means "dont type check this", object means "assume that the thing in this parameter is a descendant of `object` and don't allow using methods that `object` doesn't define.